### ISSUE TITLE
abandoned or live?

### ISSUE THREAD

**Issue by drdozer** [andrewcooke/ParserCombinator.jl#37]
Hi - I was wondering if this project is alive or has been abandoned? I wanted to try to use a combinator-based parser in julia and found this one. However, there haven't been any git commits for a while. I've hit a snag where StarPlus tends to get stuck in stupidly deep recursions, and have also found the error reporting to be a bit difficult to follow but don't really want to end up as the only user/developer of a library.

**Comment by oxinabox** [comment]
Abandoned.
I think it kinda works still, maybe?

There was however a workshop on parsering just last week at JuliaCon.
Which can no doubt give good recommendations as to what the current tools are.
I still haven't watched it, but i hope it is something as nice as this.
https://www.youtube.com/watch?v=RpCnP-S7txI

**Comment by drdozer** [comment]
Thanks @oxinabox. I did watch that video and had a look at his project but couldn't work out how to actually use it. I was able to get a parser up in a couple of hours with this project, before hitting performance and debugging problems. I think it's due to backtracking but without better debug tooling I can't really work out what to do to fix things.

My dream is of a parsing environment where i can fire up a notebook session and start writing a parser against a library of inputs and get real-time parse graphs plotted for me and highlighed parsed text to figure out what's going on.

**Comment by alyst** [comment]
FYI, in the absence of @andrewcooke we finally managed to release the new version (2.1.0), which should work with the recent Julia and packages.
But I just did it as an effort to revive *GraphIO.jl*, I don't have plans to maintain/extend *ParserCombinator.jl*.

**Comment by drdozer** [comment]
@alyst  thanks - did you release it here or on a forked github project?

Julia has the potential to enable really efficient combinator parsers, as the method overloading would let us rewrite easy to understand but inefficient combinations of primitives in to efficient but difficult to understand ones. But like you, I don't have the hours to maintain it.

**Comment by alyst** [comment]
@drdozer No, it's not forked, it's the very same package. It's possible for package "collaborators" to trigger the release via juliahub.org.
